api/src/test/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProviderTest.java

```diff
@@ -331,4 +332,41 @@ public void deleteEncounter_shouldThrowResourceNotFoundExceptionWhenIdRefersToNo
 		when(encounterService.delete(WRONG_ENCOUNTER_UUID)).thenReturn(null);
 		resourceProvider.deleteEncounter(new IdType().setValue(WRONG_ENCOUNTER_UUID));
 	}
+
+	@Test
+	public void searchForEncounters_shouldReturnEncounterEverything() {
+		when(encounterService.getEncounterEverything(any()))
+				.thenReturn(new MockIBundleProvider<>(Collections.singletonList(encounter), 10, 1));
+
+		IBundleProvider results = resourceProvider.getEncounterEverything(new IdType(ENCOUNTER_UUID));
+
+		List<IBaseResource> resultList = getAllResources(results);
+
+		assertThat(resultList, notNullValue());
+		assertThat(resultList.size(), equalTo(1));
+		assertThat(resultList.get(0).fhirType(), equalTo(FhirConstants.ENCOUNTER));
+		assertThat(((Encounter) resultList.iterator().next()).getId(), equalTo(ENCOUNTER_UUID));
+	}
+
+	@Test
+	public void searchForEncounters_shouldReturnNullForEncounterEverythingWhenIdParamIsMissing() {
+		IBundleProvider results = resourceProvider.getEncounterEverything(null);
+		assertThat(results, nullValue());
+	}
+
+	@Test
+	public void searchForEncounters_shouldReturnNullForEncounterEverythingWhenIdPartIsMissingInIdParam() {
+		IBundleProvider results = resourceProvider.getEncounterEverything(new IdType());
+		assertThat(results, nullValue());
+	}
+
+	@Test
+	public void searchForEncounters_shouldReturnNullEncounterEverythingWhenIdPartIsEmptyInIdParam() {
+		IBundleProvider results = resourceProvider.getEncounterEverything(new IdType(""));
+		assertThat(results, nullValue());
+	}
+
+	private List<IBaseResource> getAllResources(IBundleProvider result) {
```

Comment:
thanks @Medhavi-16    do you really need to add this method under the tests?

Reply:
@Medhavi-16 I'm also wondering this.

Reply:
@HerbertYiga @ibacher I'm sorry, I've been M.I.A.. Yeah, we don't need this method. I'll remove it.

Reply:
@ibacher PTAL.